test/rulesets/Random/random.jl

```diff
@@ -19,4 +19,27 @@
             @test all(map(x -> x isa Zero, pb(10)))
         end
     end
+
+    @testset "rand" begin
+        non_differentiables = [((), Float64),
+                               ((Random.default_rng(),), Float64),
+                               ((Random.default_rng(),2,2), Matrix{<:Float64}),
+                               ((Float32,), Float32),
+                               ((Float32,2,2), Matrix{<:Float32}),
+                               ((Float32,(2,2)), Matrix{<:Float32}),
+                               ((2,2), Matrix{<:Float64})]
+
+        for (args,xType) in non_differentiables
+            x, dΩ = frule((), rand, args...)
+            @test x isa xType
+            @test dΩ isa DoesNotExist
+
+            x, pb = rrule(rand, args...)
+            @test x isa xType
+            @test first(pb(10)) isa Zero
```

Comment:
Isn't it important to test not the first partial, but rather all the others?
```suggestion
            dself, dargs = Iterators.peel(pb(10.0))
            @test dself isa Zero
            @test all(darg isa DoesNotExist for darg in dargs)
```

Reply:
yes, makes sense! still learning how everything works here :)